Enable minification in production builds

HarkiratS1511 opened the pull request:
Fixes #7736 

## Description
Enables minification in production webpack builds by changing `minimize: false` to `minimize: true` in `app/webpack.prod.config.js`.

## Problem
The published frontend assets were ~63MB compared to JupyterLab's ~17MB, despite using the same underlying technology. Investigation revealed that the production webpack configuration explicitly disabled minification.

## Changes
- Changed `minimize: false` to `minimize: true` in `app/webpack.prod.config.js` (line 18)

## Expected Impact
- Reduces asset size from ~63MB to ~17MB (approximately 73% reduction)
- Improves page load times
- Reduces bandwidth usage
- Faster pip/conda installations

## Testing
Unable to test locally due to development environment constraints. The change is straightforward (single boolean flag), and CI will validate the build. Ready for maintainer review.

HarkiratS1511 committed:
Enable minification in production builds

Fixes #7736

- Changed minimize: false to minimize: true in webpack.prod.config.js
- This should reduce asset size from ~63MB to ~17MB
- Matches JupyterLab's optimized build approach

krassowski reviewed (commented):
> - Reduces asset size from ~63MB to ~17MB (approximately 73% reduction)
> - Improves page load times
> - Reduces bandwidth usage

:tada: 

> Faster pip/conda installations

Does it? It looks like the release artifacts actually ended up being larger than on `main`:


<details>

<img width="1031" height="490" alt="image" src="https://github.com/user-attachments/assets/526f55d7-a86d-4b37-bdc0-9f0002a47995" />

<img width="1031" height="490" alt="image" src="https://github.com/user-attachments/assets/6b427274-3370-4302-96d7-229e784e6177" />

<img width="781" height="558" alt="image" src="https://github.com/user-attachments/assets/1bf0b2b1-c45e-44b6-b507-f9bcbcdd64e0" />

<img width="781" height="558" alt="image" src="https://github.com/user-attachments/assets/c4343416-dd3a-49de-a309-d2174603ec47" />

This is because the map files take up more space now, for example:

<img width="565" height="212" alt="image" src="https://github.com/user-attachments/assets/ce39e39e-92c4-4cc4-aaba-c1172e7b3b78" />

<img width="565" height="212" alt="image" src="https://github.com/user-attachments/assets/ebd48fda-64b9-4409-a6cf-06795e47133a" />

</details>

I wonder if there are some options for the minimizer/terser we should copy from JupyterLab (although I also think we should update those in JupyterLab as well): https://github.com/jupyterlab/jupyterlab/blob/21f9c73298d7a450dba9927d404c87aae3011f7a/jupyterlab/staging/webpack.prod.minimize.config.js#L21-L35

jtpio commented:
Also the latest commit on `main` shows the following numbers:

https://github.com/jupyter/notebook/actions/runs/18644190793

| | |
|---|---|
| <img width="1070" height="328" alt="image" src="https://github.com/user-attachments/assets/040aa443-0c0b-4009-92c1-906cd8ea20c2" /> | <img width="944" height="88" alt="image" src="https://github.com/user-attachments/assets/a4b6ee48-1a2c-4163-93d4-29242e11fde3" /> |


While with this PR the numbers are slightly bigger?

https://github.com/jupyter/notebook/actions/runs/18612834945?pr=7748


| | |
|---|---|
| <img width="1090" height="334" alt="image" src="https://github.com/user-attachments/assets/1faadc35-5d0c-49f7-b306-05de34667827" /> | <img width="934" height="82" alt="image" src="https://github.com/user-attachments/assets/0c5a8547-ac45-4c41-9d2c-0cfd8cfc9dbb" />|

krassowski commented:
@jtpio yes this is due to map files increasing in size, see the collapsed details in my comment.

HarkiratS1511 committed:
Add TerserPlugin configuration from JupyterLab

Copied terser configuration from JupyterLab to optimize both JS and source maps.
Unable to test locally due to lib0/TypeScript compatibility issues, relying on CI for validation.

krassowski commented:
@HarkiratS1511 if you face build issues feel free to ask on jupyter zulip chat or discourse maybe we could help out. What you describe could have happened if using say `npm` over `yarn` (or an older version of `yarn`) and thus ignoring the lockfile which resolves to versions that are known to work.

As for testing you can download the built artifact from CI it's the result of Check Release workflow on the summary apge e.g. for latest commit here https://github.com/jupyter/notebook/actions/runs/18709256519?pr=7748

HarkiratS1511 committed:
Merge branch 'main' into fix-minification-7736

HarkiratS1511 commented:
I see the artifact size went up to 30.6 MB instead of down.
Looking at the config, I'm using the same TerserPlugin settings as JupyterLab, but the results are different. I'm wondering if the issue might be with how source maps are being generated - the config has devtool: 'source-map' which generates full source maps.
Do you think it's worth trying to either remove source maps entirely (devtool: false) or use a different source map strategy? Or is there something else in the build configuration I should be looking at?
